cl_dll/hud.cpp

```diff
@@ -301,6 +301,38 @@ void __CmdFunc_Append()
 	EngineClientCmd(gEngfuncs.Cmd_Argv(1));
 }
 
+void __CmdFunc_Savemap()
+{
+	FILE* saved_maps;
+	char map_name[64];
+	char map_name_to_check[64];
+
+	get_map_name(map_name, ARRAYSIZE(map_name));
+
+	if (map_name[0])
+	{
+		sprintf(map_name, "%s\n", map_name);
+		saved_maps = fopen("saved_maps.txt", "r");
+
+		if (saved_maps)
+		{
+			while (fgets(map_name_to_check, ARRAYSIZE(map_name_to_check), saved_maps))
+			{
+				if (!strcmp(map_name, map_name_to_check))
+				{
+					gEngfuncs.Con_Printf("Current map is already in saved_maps.txt");
+					fclose(saved_maps);
+					return;
+				}
+			}
+		}
+
+		saved_maps = fopen("saved_maps.txt", "a");
```

Comment:
Wouldn't this leak the file opened by previous `fopen` call?

Reply:
Good point. Also when opening the file for a second time, the result needs to be checked too.

I suggest not opening the file twice, but opening it once for read+write and then seeking to the end to write.

Reply:
> Wouldn't this leak the file opened by previous `fopen` call?

If im correct theres now no possibility of such leak occuring, because fclose() doesn't need to be called after opening a file in r+ mode that doesn't exist.